tests/v2/validation/upgrade/kubernetes_test.go

```diff
@@ -106,4 +110,23 @@ func (u *UpgradeKubernetesTestSuite) testUpgradeSingleCluster(clusterName, versi
 		u.T().Logf("[%v]: Validating updated cluster's nodepools kubernetes versions", clusterName)
 		validateNodepoolVersions(u.T(), client, updatedCluster, version, !isCheckingCurrentCluster)
 	}
+
+	err = nodestat.IsNodeReady(client, clusterMeta.ID)
+	require.NoError(u.T(), err)
+
+	clusterToken, err := clusters.CheckServiceAccountTokenSecret(client, clusterName)
+	require.NoError(u.T(), err)
+	assert.NotEmpty(u.T(), clusterToken)
+
+	podResults, podErrors := pods.StatusPods(client, clusterMeta.ID)
+	assert.NotEmpty(u.T(), podResults)
+	assert.Empty(u.T(), podErrors)
+
+	if psact == string(RancherPrivileged) || psact == string(RancherRestricted) {
+		err = psadeploy.CheckPSACT(client, clusterName)
+		require.NoError(u.T(), err)
+
+		_, err = psadeploy.CreateNginxDeployment(client, clusterMeta.ID, psact)
+		require.NoError(u.T(), err)
+	}
```

Comment:
```suggestion
	if psact == string(RancherPrivileged) || psact == string(RancherRestricted) {
		err = psadeploy.CheckPSACT(client, clusterName)
		require.NoError(u.T(), err)

		_, err = psadeploy.CreateNginxDeployment(client, clusterMeta.ID, psact)
		require.NoError(u.T(), err)
	}
	
	podResults, podErrors := pods.StatusPods(client, clusterMeta.ID)
	assert.NotEmpty(u.T(), podResults)
	assert.Empty(u.T(), podErrors)
```

This is the second part of what I was asking, wouldn't the pod status check cover nginx deployment as well?

Reply:
It can, we had that check how it is because we want to check the pods in the system workload page as part of one of the team's general validation checks. Since the PSACT check already checks the state, technically it does that.

It's not a big deal to me either way to put it after, but I'd have to do that for all the other files too which would make the PR a bit bigger

Reply:
Resolved. Made the change to this file and the other provisioning files that need the change.